List games

hathcode opened the pull request:
Need a good eye to check over my changes.  Had to resolve some conflicts when I pulled recent code changes into my working branch.  I think I got it right!

cbohara commented:
Are we supposed to have this is app/models/game.rb ?

 +  def index
 +
 +  end

hathcode commented:
Corrected, thanks for catching that!

jacobwgillespie commented on db/migrate/20160711185930_alter_games_add_user_id_column.rb:
```diff
@@ -0,0 +1,6 @@
+class AlterGamesAddUserIdColumn < ActiveRecord::Migration
```
We likely don't need this relation, as there's already `white_user_id` and `black_user_id` for the two users.

kyidkim commented on app/models/game.rb:
```diff
@@ -2,4 +2,9 @@ class Game < ActiveRecord::Base
   belongs_to :black_user, :class_name => 'User'
   has_many :pieces
+
+  scope :needing_second_player, -> { where(black_user_id: nil) }
```
@jacobwgillespie  is this another instance where we don't need to include "_id" when calling the user?

jacobwgillespie replied:
I'm about 80% sure just "black_user" would work as Rails would translate that into "black_user_id" but this definitely works since it's explicit.

hathcode commented:
Dadgum, I meant to go back and remove the user-id from the game model.  I added it in there to get things to work in a "general case" and only got it to work using the color_user right before I pushed.  I removed it from game.rb, but forgot to remove from the model.

Also, i couldn't get it to load correctly when i used black_user, it only worked with "black_user_id."  I'll go back and check it again and make the adjustment if it works.

hathcode commented:
Ok, so I took off "id" in the scope, and it still works fine.  I likely had made another change when I was checking that and threw myself a false positive.  At any rate, cleaning it up (again), removing user id, and will upload shortly.

hathcode commented:
All cleaned up.  Thanks for the input guys.

cbohara commented:
👍

cbohara commented:
hey @hathcode i pulled down master and still getting an error raised for <h4>White Player:  <%= game.white_user.email %></h4>

cbohara commented:
Were you able to get it working on your local machine?
